### DISCUSSION TITLE
sourcemaps not being generated

### DISCUSSION THREAD

**Discussion by mwoodpatrick** [vercel/next.js#28897] [Help]
I have an nx monorepo which has a nextjs app (in typescript)

If I run

   npx nx build site

then this command mostly respects the setting of:

productionBrowserSourceMaps

in next.config.js except that it does not generate map files for any of my files in the api subdirectory, I would expect that with this setting it would generate source maps for all source files. Does anyone know why it would skip these files or best way to debug?

However,  when I run:

    npx nx serve site

it does not appear to generate any map files. I would expect it to always generate source maps.

Does anyone know why this is happening or best way to debug?

**Comment by thomascouto** [comment]
Same here, what am I missing ?

```
/** @type {import('next').NextConfig} */
const nextConfig = {
	output: 'export',
	productionBrowserSourceMaps: true,
	experimental: {
		appDir: true,
		serverSourceMaps: true,
	}
}

module.exports = nextConfig
```

**Comment by maro1993** [comment]
Ist this only supported for the app router? I can not find anything regarding the serverSourceMaps property in the pages router docs. In the app router docs i atleast find this config property mentioned in this section https://nextjs.org/docs/app/building-your-application/optimizing/memory-usage#disable-source-maps .

As a workaround you might give the webpack based solution in this blog post a shot.

https://www.highlight.io/blog/lw5-next-js-server-sourcemaps
